Frontends/glslang/GlslangToTopVisitor.cpp

```diff
@@ -1391,7 +1391,12 @@ bool TGlslangToTopTraverser::visitAggregate(glslang::TVisit visit, glslang::TInt
         result = createUnaryIntrinsic(node->getOp(), precision, operands.front());
         break;
     default:
-        result = createIntrinsic(node->getOp(), precision, uniform, operands, glslangOperands.front()->getAsTyped()->getBasicType() == glslang::EbtUint);
+        // Check first for intrinsics that can be done natively
+        if (node->getOp() == glslang::EOpMix && 
+              gla::GetBasicTypeID(operands.front()) == llvm::Type::IntegerTyID)
+            result = llvmBuilder.CreateSelect(operands[2], operands[1], operands[0]);
```

Comment:
Are these all properly smeared or other to handle GLSL mixing of vectors and scalars?